sky/provision/kubernetes/network_utils.py

```diff
@@ -19,9 +20,13 @@
 def get_port_mode(
         mode_str: Optional[str] = None) -> kubernetes_enums.KubernetesPortMode:
     """Get the port mode from the provider config."""
-    mode_str = mode_str or skypilot_config.get_nested(
-        ('kubernetes', 'ports'),
-        kubernetes_enums.KubernetesPortMode.LOADBALANCER.value)
+    curr_context = kubernetes_utils.get_current_kube_config_context_name()
+    running_kind =  curr_context == kubernetes_utils.KIND_CONTEXT_NAME
+    default = (
+        (running_kind and kubernetes_enums.KubernetesPortMode.INGRESS.value) or
+        kubernetes_enums.KubernetesPortMode.LOADBALANCER.value
+    )
+    mode_str = mode_str or skypilot_config.get_nested(('kubernetes', 'ports'), default)
```

Comment:
Thanks @shethhriday29! As we discussed, can we remove this and instead have the ground truth for the `port_mode` to be stored in `~/.sky/config.yaml`? 

Note that in doing so, we must backup the user's current config when `create_cluster.sh` is called, and restore it when `delete_cluster.sh` is called.

I.e., In `create_cluster.sh`, we read ~/.sky/config.yaml and :

1. Backup `kubernetes.ports` key by commenting it with a unique identifier like so (if it doesn't exist, backup with `port: loadbalancer`):
```
# ==== Backup made by sky local up ====
# kubernetes:
#   port:  loadbalancer/ingress
```
2. Write `kubernetes.port = ingress`

In `delete_cluster.sh`, we restore the backup if it exists.